### PR TITLE
Create xor_swap.cpp

### PR DESCRIPTION
**Pull request by Vaibhav1671998**
added code to swap two numbers using xor operator in c++

### CODE REVIEW

**Comment by Vaibhav1671998** [comment]
check this one out @arnavb

File: code/bit_manipulation/xor_swap/xor_swap.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include < iostream >
+using namespace std;
+
+void xor_swap(int * a, int * b) {
+    * a = * a ^ * b;
```
Comment (arnavb) [inline code/bit_manipulation/xor_swap/xor_swap.cpp]: No spaces with the asterisks.

File: code/bit_manipulation/xor_swap/xor_swap.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    * b = * a ^ * b;
+    * a = * a ^ * b;
+
+  return;
```
Comment (arnavb) [inline code/bit_manipulation/xor_swap/xor_swap.cpp]: Not necessary

File: code/bit_manipulation/xor_swap/xor_swap.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+int main() {
```
Comment (arnavb) [inline code/bit_manipulation/xor_swap/xor_swap.cpp]: Brace on it’s own line.

File: code/bit_manipulation/xor_swap/xor_swap.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  int a = 10, b = 15;
+
+  cout << "Before swapping: A = " << a << " and B = " << b << "\n";
+
+  xor_swap( & a, & b);
```
Comment (arnavb) [inline code/bit_manipulation/xor_swap/xor_swap.cpp]: No spaces with the ampersand and between the parentheses

**Comment by Vaibhav1671998** [comment]
Done with the changes @arnavb

**Comment by Vaibhav1671998** [comment]
Plzz merge this @AdiChat

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**